spinta/config/__init__.py

```diff
@@ -74,115 +73,106 @@
     'debug': False,
 }
 
-TEMPLATES = {
-    ('backends',): 'default',
-    ('manifests',): 'default',
-}
 
+class Config:
+
+    def __init__(self):
+        self._config = {
+            'cliargs': {},
+            'environ': {},
+            'envfile': {},
+            'default': {},
+        }
+        self.add(CONFIG)
+
+    def add(self, config):
+        self._config['default'].update(_traverse(config))
+        self._config['default'].update(_get_inner_keys(self._config['default']))
+
+    def add_cli_args(self, args):
+        for arg in args:
+            name, value = arg.split('=', 1)
+            key = tuple(name.split('.'))
+            self._config['cliargs'][key] = value
+        self._config['cliargs'].update(_get_inner_keys(self._config['cliargs']))
+
+    def add_env_file(self, path):
+        if isinstance(path, str):
+            path = pathlib.Path(path)
+
+        if not path.exists():
+            return
+
+        with path.open() as f:
+            for line in f:
+                if line.startswith('#'):
+                    continue
+                line = line.strip()
+                if line == '':
+                    continue
+                if '=' not in line:
+                    continue
+                name, value = line.split('=', 1)
+                self._config['envfile'][name] = value
+
+    def set_env(self, environ):
```

Comment:
In this case, environment variables are not added, but are set. All other `add*` methods do add things on top of existing config vars. But in environment case, I don't want to make copy of `os.environ`, so instead of modifying it in place, I just add it.

So naming here is correct, because it reflects real behavior which is different from other `add*` methods.

Reply:
agreed.